server/order/controller/product_controller.js

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2017 The Absolute Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import {Product} from '../model/product';
+
+/**
+ * Add product data what admin added.
+ * @param {json} productData data what admin sent.
+ * @return {promise} result product data what admin sent.
+ */
+export function addProduct(productData) {
+  return new Promise((resolve, reject) => {
+    Product.findOneAndUpdate({name: productData.name}
```

Comment:
As far as I know, the mongoose already supports Promise as follows:
  - http://mongoosejs.com/docs/promises.html

Reply:
Mongoose queries are not promises. However, they do have a .then() function for yield and async/await. If you need a fully-fledged promise, use the .exec() function.

Reply:
Yeah, but you can just use then() or exec() instead.
IMHO, wrapper seems unnecessary, do you agree with me?